packages/api/cms-api/src/user-permissions/content-scope.service.ts

```diff
@@ -22,61 +24,72 @@ export class ContentScopeService {
-                        throw new Error("Content Scope from arg doesn't match affectedEntity scope, usually you only need one of them");
+                const row = await repo.findOneOrFail(args[affectedEntity.options.idArg]);
+                if (row.scope) {
+                    contentScope = row.scope;
+                } else {
+                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                    const scoped = this.reflector.getAllAndOverride<ScopedEntityMeta>("scopedEntity", [
+                        affectedEntity.entity as EntityClass<unknown>,
+                    ]);
+                    if (!scoped) {
+                        return undefined;
                     }
+                    contentScope = await scoped.fn(row);
+                }
+            } else if (affectedEntity.options.pageTreeNodeIdArg && args[affectedEntity.options.pageTreeNodeIdArg]) {
+                if (!args[affectedEntity.options.pageTreeNodeIdArg]) {
+                    throw new Error(`${affectedEntity.options.pageTreeNodeIdArg} arg not found`);
                 }
-                return contentScope;
+                if (this.pageTreeService === undefined) {
+                    throw new Error("pageTreeNodeIdArg was given but no PageTreeModule is registered");
+                }
+                const node = await this.pageTreeService.createReadApi({ visibility: "all" }).getNode(args[affectedEntity.options.pageTreeNodeIdArg]);
+                if (!node) throw new Error("Can't find pageTreeNode");
+                contentScope = node.scope;
+            } else {
+                // TODO implement something more flexible that supports something like that: @AffectedEntity(Product, ProductEntityLoader)
+                throw new Error("idArg or pageTreeNodeIdArg is required");
             }
+            if (contentScope === undefined) throw new Error("Scope not found");
             if (args.scope) {
-                return args.scope;
+                // args.scope also exists, check if they match
+                if (!isEqual(args.scope, contentScope)) {
+                    throw new Error("Content Scope from arg doesn't match affectedEntity scope, usually you only need one of them");
+                }
             }
+            return contentScope;
+        }
+        if (args.scope) {
+            return args.scope;
+        }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    private async getArgs(context: ExecutionContext): Promise<Record<string, any>> {
+        if (context.getType().toString() === "graphql") {
+            return GqlExecutionContext.create(context).getArgs();
+        } else {
+            const request = context.switchToHttp().getRequest();
+            // Execute multer to process multipart/form-data requests because NestJS hasn't done it yet
+            const postMulterRequest: Request = await new Promise((resolve, reject) => {
```

Comment:
What about "normal" REST requests that are not multipart/form-data?

> Multer cannot process data which is not in the supported multipart format (multipart/form-data). Also, note that this package is not compatible with the FastifyAdapter.

https://docs.nestjs.com/techniques/file-upload

Reply:
I updated the code a bit so that all args from the controller are passed. b611a23fd754477c8f383ac639a5000fcce75ad0